packages/java-parser/src/unicode.js

```diff
@@ -0,0 +1,144 @@
+"use strict";
+const unicode = {};
+const fs = require("fs");
+const path = require("path");
+const args = process.argv.slice(2)[0];
+let lines;
+try {
+  lines = fs.readFileSync(args, "utf-8").split("\n");
+} catch (err) {
+  throw Error("Please specify the path of the UnicodeData.txt file");
+}
+// The Categories we only want to parse from the file.
+// We don't need to store characters from the UnicodeData file that we are not going to use
+// Only adding characters that are accepted as first character of a Java Identifier. See specs below.
+// https://docs.oracle.com/javase/7/docs/api/java/lang/Character.html#isJavaIdentifierStart(char)
+const firstIdentCharCategories = new Set([
+  "Ll",
+  "Lm",
+  "Lo",
+  "Lt",
+  "Lu",
+  "Nl",
+  "Sc",
+  "Pc"
+]);
+
+// Adding remaining categories that can be used in a java identifier.
+// https://docs.oracle.com/javase/7/docs/api/java/lang/Character.html#isJavaIdentifierPart(char)
+const restIdentUnicodeCategories = new Set(["Mn", "Cf"]);
+
+const manuallyAddedCharacters = new Set([]);
+// Below adding the isIdentifierIgnorable characters from Java Specs but not from FORMAT general category
+for (let i = 0; i < 8; i++) {
+  manuallyAddedCharacters.add(i);
+}
+for (let i = parseInt("000E", 16); i < parseInt("001B"); i++) {
```

Comment:
Why do we need so many tiny for loops?
The only thing different between the loops is the different ranges.

```javascript
const ranges = [
  {start: " "000E", end: "001B"},
  {start: " "007F", end: "009F"},
// ...
]

ranges.forEach((range) => {
for (let i = parseInt(range.start, 16); i < parseInt(range.end); i++) {
  manuallyAddedCharacters.add(i);
}
})
```